tests/test_views.py

```diff
@@ -27,6 +27,16 @@ def test_auth_view_retains_next_url(client):
     assert client.session[REDIRECT_SESSION_FIELD_NAME] == '/go-here-after-logging-in/'
 
 
+@pytest.mark.django_db
+def test_auth_view_retains_unsafe_next_url(client):
+    url = reverse('authbroker:login') + '?next=https://danger.com'
+    response = client.get(url)
+    assert response.status_code == 302
+    assert AUTHORISATION_URL in response.url
+    assert 'code=someCode' not in response.url
```

Comment:
maybe you meant `assert 'next=https://danger.com' not in response.url` ?

Reply:
Nah, that was just an artifact from the test I copied. I've removed it.